vertexy/src/public/ConstraintTypes.h

```diff
@@ -500,6 +500,16 @@ class GraphCulledVector
 		}
 		return out;
 	}
+
+	static GraphCulledVector allOptional(vector<T> vec)

```

Comment:
pass as `const vector<T>&`

Reply:
Also, the `allOptional` that takes two iterators was intended as the more generic version of this (that can take any STL style container instead of only a vector). I.e. call `allOptional(vec.begin(), vec.end())`